prow/gcpproxy-coverage.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+# Copyright 2019 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Presubmit script triggered by Prow.
+
+# Fail on any error.
+set -eo pipefail
+
+gcloud config set core/project cloudesf-testing
+
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+cd "${ROOT}"
+. ${ROOT}/scripts/all-utilities.sh || { echo 'Cannot load Bash utilities';
+exit 1; }
+
+echo '======================================================='
+echo '===================== Setup Cache ====================='
+echo '======================================================='
+try_setup_bazel_remote_cache "${PROW_JOB_ID}" "${IMAGE}" "${ROOT}" "${JOB_TYPE}-coverage"
+
+
+echo '======================================================='
+echo '==================== C++ Coverage ====================='
+echo '======================================================='
+. ${ROOT}/third_party/tools/coverage/cpp_unit.sh
+
+echo '======================================================='
+echo '=================== Upload Coverage ==================='
+echo '======================================================='
+PUBLIC_DIRECTORY=""
```

Comment:
should we key with SHA so we will store coverage for each SHA,  
it may be useful if we want to investigate a code coverage problem?

Reply:
Done

Reply:
Note that I set the GCS bucket to delete entries after 60 days